docs/data/data-grid/rows/rows.md

```diff
@@ -80,6 +82,20 @@ In addition, the area in which `onRowsScrollEnd` is called can be changed using
 
 {{"demo": "InfiniteLoadingGrid.js", "bg": "inline", "disableAd": true}}
 
+### Lazy loading [<span class="plan-pro"></span>](https://mui.com/store/items/mui-x-pro/)
+
+To allow the grid to lazy load data, set `rowsLoadingMode="server"`.
+Then the `rowCount` needs to be set and the number of initially loaded rows needs to be less than the `rowCount` value.
+In addition, you need to handle the `onFetchRows` callback to fetch the rows for the corresponding index.
+Finally, you need to use the `apiRef.current.insertRows()` to tell the DataGrid where to insert the newly fetched rows.
+
+:::info
+In order for the filtering and sorting to work you need to set their modes to `server`.
+You can find out more information about how to do that on the [server-side filter page](/components/data-grid/filtering/#server-side-filter) and on the [server-side sorting page](/components/data-grid/sorting/#server-side-sorting).
+:::
+
+{{"demo": "LazyLoadingGrid.js", "bg": "inline", "disableAd": true}}
```

Comment:
If all the rows are loaded the `onFetchRows` won't be called.
One optimization that is needed by the user is to implement throttle because now this callback is being called a lot while scrolling.

Reply:
It seems to fetch for each row that becomes visible in the viewport. In practice, it's probably not usable in production in the current shape as it would flood the server API.